src/model/application/mod.rs

```diff
@@ -45,43 +45,56 @@ pub struct CurrentApplicationInfo {
     pub icon: Option<ImageHash>,
     pub description: FixedString,
     #[serde(default)]
-    pub rpc_origins: FixedArray<String>,
+    pub rpc_origins: FixedArray<FixedString>,
     pub bot_public: bool,
     pub bot_require_code_grant: bool,
     #[serde(default)]
+    pub bot: User,
+    #[serde(default)]
     pub terms_of_service_url: Option<FixedString>,
     #[serde(default)]
     pub privacy_policy_url: Option<FixedString>,
-    pub owner: Option<User>,
-    // omitted `summary` because it deprecated
+    #[serde(default)]
+    pub owner: User,
     pub verify_key: FixedString,
     pub team: Option<Team>,
     #[serde(default)]
     pub guild_id: Option<GuildId>,
     #[serde(default)]
+    pub guild: Option<PartialGuild>,
+    #[serde(default)]
     pub primary_sku_id: Option<SkuId>,
     #[serde(default)]
-    pub slug: Option<FixedString>,
+    pub slug: FixedString,
     #[serde(default)]
-    pub cover_image: Option<FixedString>,
+    pub cover_image: FixedString,
```

Comment:
Similar to the application icon, the API returns a hash of a constant size, so this should use [`ImageHash`](https://serenity-rs.github.io/serenity/next/serenity/model/misc/struct.ImageHash.html) to store the hash inline. Also should be wrapped in an `Option<...>`.